tests/jballerina-unit-test/src/test/resources/test-src/action/client_resource_access_action.bal

```diff
@@ -612,6 +612,30 @@ function testClosuresFromPathParams() {
     assertEquality(e, 12);
 }
 
+public type Params record {|
+    never headers?;
+    int...;
+|};
+
+client class MyClient10 {
+    resource function get [string... path](string? headers = (), *Params params) returns int {
+        return params.get("id");
+    }
+
+    resource function post foo/[int a]/bar(*Params params) returns int {
+        return params.get("id") + a;
+    }
+}
+
+public function testAccessingResourceWithIncludedRecordParam() {
+    MyClient10 cl = new;
+    int a = cl->/foo/bar(id = 1);
+    assertEquality(a, 1);
+
+    int b = cl->/foo/[4]/bar.post(id = 1);
+    assertEquality(b, 5);
+}
```

Comment:
We can add more tests, right?

e.g., 
1. calling the first method with an argument for `headers` 
2. passing a record value for `params` instead of passing named args for the fields both as a positional arg and as a named arg